spark/src/main/scala/org/apache/spark/sql/delta/PreDowngradeTableFeatureCommand.scala

```diff
@@ -213,6 +213,56 @@ case class VacuumProtocolCheckPreDowngradeCommand(table: DeltaTableV2)
   }
 }
 
+case class ManagedCommitPreDowngradeCommand(table: DeltaTableV2)
+  extends PreDowngradeTableFeatureCommand
+  with DeltaLogging {
+
+  /**
+   * We disable the feature by removing the following table properties:
+   *    1. DeltaConfigs.MANAGED_COMMIT_OWNER_NAME.key
+   *    2. DeltaConfigs.MANAGED_COMMIT_OWNER_CONF.key
+   *    3. DeltaConfigs.MANAGED_COMMIT_TABLE_CONF.key
+   *
+   * @return true if any change to the metadata (the three properties listed above) was made.
+   *         false otherwise.
+   */
+  override def removeFeatureTracesIfNeeded(): Boolean = {
```

Comment:
This function starts returning false as soon as these 3 table properties are not part of the table Metadata. That signals that there are no traces of the feature - which is not true in case there are unbackfilled commits in the Snapshot.

Once this method returns false, the AlterTable command will do a commit to remove managed-commit from the table and that takes care of backfilling any unbackfilled commits. So although it doesn't shouldn't cause any issue but semantically we should not give a green light from here until the commits are also backfilled.

Could we make sure that unbackfilled commits are backfilled by doing:

        table.initialSnapshot.deltaLog.update().ensureCommitFilesBackfilled()

Reply:
Actually the workaround that I suggested `snapshot.ensureCommitFilesBackfilled` won't work as there is no commit-owner on the table after `AlterTableUnsetPropertiesDeltaCommand` is done .

Reply:
Okay, this sounds reasonable. Semantically, `removeFeatureTracesIfNeeded` is not complete if there are any unbackfilled commits after it has finished.